cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/ExpressionFactory.java

```diff
@@ -18,20 +22,42 @@
     private static final Pattern PARENS = Pattern.compile("\\(([^)]+)\\)");
     private static final Pattern ALPHA = Pattern.compile("[a-zA-Z]+");
     private final ParameterTypeRegistry parameterTypeRegistry;
+    private final ParameterTypeRegistryTableConverter tableConverter;
 
     public ExpressionFactory(ParameterTypeRegistry parameterTypeRegistry) {
         this.parameterTypeRegistry = parameterTypeRegistry;
+        this.tableConverter = new ParameterTypeRegistryTableConverter(parameterTypeRegistry);
     }
 
-    public Expression createExpression(String expressionString, Type tableType) {
-        return createExpression(expressionString, parameterTypeRegistry.lookupTableTypeByType(tableType));
+    public Expression createExpression(String expressionString, final Type tableType) {
+        if (tableType == null) throw new CucumberExpressionException("tableType can not be null");
+
+        return createExpression(expressionString, new Transformer<List<List<String>>, Object>() {
```

Comment:
We can extract the `Transformer` and leave its creation to the caller. If we also provide the default table transform we  can cut the dependency between datatable and cucumber-expressions.